src/sys/output.cxx

```diff
@@ -153,4 +155,21 @@ ConditionalOutput output_error(Output::getInstance());
 ConditionalOutput output_verbose(Output::getInstance(), false);
 ConditionalOutput output(Output::getInstance());
 
+void backupfile(const char* filename) {
+  struct stat buffer;
+  if (stat(filename, &buffer) == 0){
```

Comment:
I know `stat` is POSIX, but it's not standard C++ and we should prefer to keep to the standard. An alternative would be `std::ifstream`, which is convertable to `bool`:

```cpp
bool fileExists(const std::string& filename) {
  return static_cast<bool>(std::ifstream(filename));
}
```

If we eventually get to C++17, this could just be `std::filesystem::exists`

Reply:
std::ifstream checks whether the file is readable - while stat actually checks the existence.
Thus stat returns true for folders/symlinks/... - while std::ifstream would not 'detect' them.
Undetected symlinks could even lead to privilege escalation, while I don't think it is a issue here, I still think stat is the right tool for this, until we get C++17.

Are there hones attempts to port BOUT++ to non-posix systems?